tests/nvapi_vulkan_mocks.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include "nvapi_tests_private.h"
+#include "../src/nvapi/vk.h"
+
+using namespace trompeloeil;
+
+class VkDeviceMock {
+    MAKE_MOCK4(vkCreateSemaphore, VkResult(VkDevice, const VkSemaphoreCreateInfo*, const VkAllocationCallbacks*, VkSemaphore*));
+    MAKE_MOCK3(vkDestroySemaphore, VkResult(VkDevice, VkSemaphore, const VkAllocationCallbacks*));
+    MAKE_MOCK3(vkSetLatencySleepModeNV, VkResult(VkDevice, VkSwapchainKHR, const VkLatencySleepModeInfoNV*));
+    MAKE_MOCK3(vkLatencySleepNV, VkResult(VkDevice, VkSwapchainKHR, const VkLatencySleepInfoNV*));
+    MAKE_MOCK3(vkGetLatencyTimingsNV, VkResult(VkDevice, VkSwapchainKHR, VkGetLatencyMarkerInfoNV*));
+    MAKE_MOCK3(vkSetLatencyMarkerNV, VkResult(VkDevice, VkSwapchainKHR, const VkSetLatencyMarkerInfoNV*));
+
+    static VkResult PFN_vkCreateSemaphore(VkDevice device, const VkSemaphoreCreateInfo* pCreateInfo, const VkAllocationCallbacks* pAllocator, VkSemaphore* pSemaphore) {
```

Comment:
Using `PFN_` prefix (that's usually used for the type name) in method name feels weird. I'd drop the prefix here, and if that causes collisions with mocks above, I'd then rename mocks to include `p` prefix or drop the `vk` prefix.